src/providers/microsoft/azure/mod.rs

```diff
@@ -271,10 +271,13 @@ impl Azure {
             .context("failed to decrypt cms blob")?;
 
         // convert that to the OpenSSH public key format
-        let ssh_pubkey = crypto::p12_to_ssh_pubkey(&p12)
-            .context("failed to convert pkcs12 blob to ssh pubkey")?;
+        let ssh_pubkey = match crypto::p12_to_ssh_pubkey(&p12)
+            .context("failed to convert pkcs12 blob to ssh pubkey")? {
+                Some(key) => key,
+                None => return Ok(None)
+            };
 
-        Ok(ssh_pubkey)
+        Ok(Some(vec![ssh_pubkey]))
```

Comment:
How about making the signature of this function as `Result<Option<PublicKey>>` and then here it'd just be

```suggestion
        Ok(Some(ssh_pubkey))
```

Reply:
Actually, I think once we do the above, we can shrink this whole thing to just call `crypto::p12_to_ssh_pubkey(&p12).context("failed to convert pkcs12 blob to ssh pubkey")` as the final line of the function since it already matches the signature we need.